detection_rules/rule_validators.py

```diff
@@ -192,11 +192,17 @@ def validate_integration(
                 integration_schema_data["integration"],
             )
             integration_schema = integration_schema_data["schema"]
+            stack_version = integration_schema_data["stack_version"]
 
             # Add non-ecs-schema fields
             for index_name in data.index:
                 integration_schema.update(**ecs.flatten(ecs.get_index_schema(index_name)))
 
+            # Add custom schema fields for appropriate stack version
+            if data.index and CUSTOM_RULES_DIR:
+                for index_name in data.index:
+                    integration_schema.update(**ecs.flatten(ecs.get_custom_index_schema(index_name, stack_version)))
```

Comment:
nit: This looks identical to lines L396-L400. Might be worth moving to a small method for cleanliness.

Reply:
I agree it would be better, but since I think both validation methods have a number of nearly identical code blocks, that this would be better done in a larger refactor. 

For instance another identical block would be for non-ecs schema fields.

```
for index_name in data.index:
                integration_schema.update(**ecs.flatten(ecs.get_index_schema(index_name)))
```